plugins/by-name/glance/default.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, ... }:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "glance";
+  packPathName = "glance.nvim";
+  package = "glance-nvim";
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  settingsExample = {
+    height = 40;
+    zindex = 50;
+    border = {
+      enable = true;
+    };
+    use_trouble_qf = true;
+  };
+
+  extraConfig = cfg: {
+    warnings =
+      lib.optional
+        (cfg.settings ? use_trouble_qf && cfg.settings.use_trouble_qf && !config.plugins.trouble.enable)
```

Comment:
Actually, since this is a freeform option, we may need to be a bit more explicit:

```suggestion
        (cfg.settings ? use_trouble_qf && builtins.isBool cfg.settings.use_trouble_qf && cfg.settings.use_trouble_qf && !config.plugins.trouble.enable)
```

Not sure how we'd handle cases where it's defined and non-bool... E.g. we can't really warn if it is a `__raw` function :thinking:

Reply:
> Not sure how we'd handle cases where it's defined and non-bool... E.g. we can't really warn if it is a `__raw` function 🤔

Hmm.. I didn't think about that. But, we are at least covering the normal usage and it's not too restrictive since we're doing a warning instead of an assertion.